client/client.go

```diff
@@ -700,6 +708,7 @@ func initRPCInfo(ctx context.Context, method string, opt *client.Options, svcInf
 		// it is used to distinguish the request is a local retry request.
 		rmt.SetTag(rpcinfo.RetryTag, strconv.Itoa(retryTimes))
 	}
+	cfg.SetPayloadCodec(svcInfo.PayloadCodec)
```

Comment:
mark

Reply:
It's serviceinfo.PayloadCodec (enumeration), which may not be right in RPCConfig. This fixes it.